### PR TITLE
Remove oneMKL getrs_batch workaround

### PR DESCRIPTION
**Pull request by gardner48**
Workaround for bug in`getrs_batch` is no longer needed

### CODE REVIEW

**Review by drreynolds — APPROVED** [review]

**Comment by balos1** [comment]
Does this require a specific version of oneMKL?

File: examples/cvode/CXX_onemkl/cvRoberts_blockdiag_onemkl.cpp [hunk 1]
```diff
@@ -129,7 +129,7 @@ int main(int argc, char *argv[])
   if (argc > 3) output = (atoi(argv[3])) ? true : false;
 
   // Create an in-order GPU queue
-#if SYCL_LANGUAGE_VERSION >= 2020
+#if SYCL_LANGUAGE_VERSION >= 2020 && !defined(SUNDIALS_SYCL_2020_UNSUPPORTED)
```
Comment (balos1) [inline examples/cvode/CXX_onemkl/cvRoberts_blockdiag_onemkl.cpp:132]: Where does `SUNDIALS_SYCL_2020_UNSUPPORTED` come from?

Comment (balos1) [reply examples/cvode/CXX_onemkl/cvRoberts_blockdiag_onemkl.cpp:132]: OK, but I dont see where it is defined in CMake. Is this just passed in as a compiler flag? Should it be documented?

**Comment by gardner48** [comment]
@balos1 and @drreynolds I've finished testing the PR and it's ready for another review pass

File: cmake/SundialsTPLOptions.cmake [hunk 2]
```diff
@@ -61,6 +61,11 @@ sundials_option(ENABLE_HIP BOOL "Enable HIP support" OFF)
 # -------------------------------------------------------------
 sundials_option(ENABLE_SYCL BOOL "Enable SYCL support" OFF)
 
+sundials_option(SUNDIALS_SYCL_2020_UNSUPPORTED BOOL
+                "Disable some uses of SYCL 2020 features" OFF
```
Comment (drreynolds) [inline cmake/SundialsTPLOptions.cmake:65]: This CMake option description is perhaps overly vague.  I realize that there are space limitations, but could this be a least a bit more descriptive?

**Review by drreynolds — DISMISSED** [review]
A couple of minor comments, but otherwise this looks great.

**Review by balos1 — APPROVED** [review]

**Pull request merged**